src/components/UserActions/UserActions.ts

```diff
@@ -288,10 +288,20 @@ export class UserActions extends Component {
             if (Boolean(args.item.getElementsByClassName('CoveoViewedByCustomer').length)) {
                 return;
             }
+            const viewedByCustomerRow = document.createElement('div');
```

Comment:
That seems a bit wrong that the user action is handling the viewed by custom logic no ?

Reply:
This is how it was before though. We kindof need a "parent" element to insert the component in the result div right?
I'm curious though, what happens if users place it elsewhere in the result templates? Does this overwrite it?

Reply:
I think that's another question that one could have (i.e. if we were to change that, that's yet another story).

Also, I invite you to check the Jira linked to the addition of this feature: https://coveord.atlassian.net/browse/SFINT-2519, it seems to me that it was required.

Reply:
You can make a function on the viewedByCustomer side where you generate the required dom, no ?

Reply:
No, it's a result template component it cannot add itself to the result template (and all the others)

Also, that's quite a bigger change that you're suggesting, let's discuss about it off GitHub.

Reply:
> This is how it was before though. We kindof need a "parent" element to insert the component in the result div right?
> I'm curious though, what happens if users place it elsewhere in the result templates? Does this overwrite it?
@nlafrance-coveo
No, if you turn this option at on, it will add the ViewedByCustomer a the bottom of every result templates that doesn't have it already.

Reply:
Cleared misunderstanding with @jeremierobert-coveo which resulted in https://github.com/coveo/search-ui-extensions/pull/58/commits/7d6d02d09f344ffaad810ba8e5c4e71f1e51e5ca

@jeremierobert-coveo , if all green please approve 👍 
Thanks again for taking the time to get to the bottom of this quiproquo

Reply:
Cleared misunderstanding with @jeremierobert-coveo which resulted in https://github.com/coveo/search-ui-extensions/pull/58/commits/7d6d02d09f344ffaad810ba8e5c4e71f1e51e5ca

@jeremierobert-coveo , if all green please approve 👍 
Thanks again for taking the time to get to the bottom of this quiproquo